### PR TITLE
feat(infrastructure): IaC

### PR DESCRIPTION
**Pull request by abhi-markan**
## Introduction
Staging and Production infrastructure requirement

## Resolution
* Updated deployment script, to reflect a refactored and generic IaC.
* Refactored IaC

### CODE REVIEW

File: .github/workflows/infrastructure.yml [hunk 1]
```diff
@@ -1,76 +1,267 @@
...
+        with:
+          inlineScript: |
+            az network vnet create \
+            --name vnet-${{ env.PRODUCT }}-${{ vars.VERSION }} \
+            --address-prefix ${{ secrets.VNET_ADDRESS_PREFIX }} \
+            --subnet-name snet-${{ env.PRODUCT }}-${{ vars.VERSION }} \
+            --subnet-prefixes ${{ secrets.VNET_SUBNET_PREFIX }} \
+            --tags ${{ env.TAGS }}
+
+      - name: VNET Peer - AMI 🔀
```
Comment (IgnatG) [inline .github/workflows/infrastructure.yml:143]: Do we need this for Exips?

File: .github/workflows/infrastructure.yml [hunk 2]
```diff
@@ -1,76 +1,267 @@
...
+            --vnet-name vnet-${{ env.PRODUCT }}-${{ vars.VERSION }} \
+            --name vnet-peer-ami-${{ env.PRODUCT }}-${{ vars.VERSION }} \
+            --query peeringState)"
+
+      - name: Web app - UI 🌐
+        uses: Azure/cli@v1.0.7
+        with:
+          inlineScript: |
+            az webapp create \
+            --name app-ui-${{ env.PRODUCT }}-${{ env.TARGET }}-${{ vars.VERSION }}.azurewebsites.net \
```
Comment (IgnatG) [inline .github/workflows/infrastructure.yml:176]: Should this be something like: app-exips-ui-dev-001?

File: .github/workflows/infrastructure.yml [hunk 3]
```diff
@@ -1,76 +1,267 @@
...
+        with:
+          inlineScript: |
+            az monitor log-analytics workspace create \
+            --name log-${{ env.PRODUCT }}-${{ env.TARGET }}-${{ vars.VERSION }} \
+            --capacity-reservation-level 100 \
+            --ingestion-access Disabled \
+            --query-access Disabled \
+            --quota 0.05 \
+            --retention-time ${{ vars.LOG_RETENTION_DAY }} \
+            --sku Free \
```
Comment (IgnatG) [inline .github/workflows/infrastructure.yml:118]: Is it worth having the SKU in a var?

File: .github/workflows/infrastructure.yml [hunk 4]
```diff
@@ -1,76 +1,267 @@
...
+        run: |
+          # Set a web app's configuration
+
+          # UI
+          az webapp config set \
+          --name app-ui-${{ env.PRODUCT }}-${{ env.TARGET }}-${{ vars.VERSION }}.azurewebsites.net \
+          --always-on true \
+          --ftps-state Disabled \
+          --http20-enabled true \
+          --min-tls-version '1.2'
```
Comment (IgnatG) [inline .github/workflows/infrastructure.yml:245]: I've seen the tls version repeating a few times. Is it worth having it in a var?

File: .github/workflows/infrastructure.yml (hunk 1 above)
Comment (abhi-markan) [reply .github/workflows/infrastructure.yml:143]: Yes for SQL DB

File: .github/workflows/infrastructure.yml (hunk 3 above)
Comment (abhi-markan) [reply .github/workflows/infrastructure.yml:118]: Good shout, have introduced `LOG_ANALYTICS_PLAN`

File: .github/workflows/infrastructure.yml (hunk 4 above)
Comment (abhi-markan) [reply .github/workflows/infrastructure.yml:245]: Have introduced `TLS_VERSION`.

**Review by IgnatG — DISMISSED** [review]
LGTM

**Review by IgnatG — DISMISSED** [review]
LGTM

**Review by ttbarnes — APPROVED** [review]

**Review by IgnatG — APPROVED** [review]

**Review by ttbarnes — DISMISSED** [review]

**Review by ttbarnes — APPROVED** [review]

**Review by ttbarnes — APPROVED** [review]

**Pull request merged**